src/encoded/tests/test_batch_download.py

```diff
@@ -133,11 +136,13 @@ def test_batch_download_lookup_column_value(lookup_column_value_item, lookup_col
 
 
 @pytest.mark.parametrize("test_input,expected", [
-    (file_type_param_list(exp_file_1, param_list_2), True),
-    (file_type_param_list(exp_file_1, param_list_1), True),
-    (file_type_param_list(exp_file_2, param_list_1), False),
+    (files_prop_param_list(exp_file_1, param_list_2), True),
+    (files_prop_param_list(exp_file_1, param_list_1), True),
+    (files_prop_param_list(exp_file_2, param_list_1), False),
+    (files_prop_param_list(exp_file_3, param_list_3), True),
+    (files_prop_param_list(exp_file_1, param_list_3), False),
```

Comment:
Would you mind adding a test for weird file_types like `bigBed bed3+`.

Reply:
I’m sure I could. Let me know what I have to do. I have a CSS change to put in as well.

Reply:
Also I don't think the files_prop_param_list function call in the parametrize decorator is a good pattern to use. The function will be evaluated at decoration time instead of test time and it causes a bunch of duplicate files_prop_param_list strings. Instead just pass parameters and call the function in the test: 
```
@pytest.mark.parametrize("exp_file, param_list, expected", [
    (exp_file_1, param_list_2, True),
    (exp_file_1, param_list_1, True),
])
def test_files_prop_param_list(exp_file, param_list, expected):
    assert file_type_param_list(exp_file, param_list) == expected
```

Reply:
> Let me know what I have to do.
Since the urlencoding is the main issue I think it needs more than unit test. 

You should actually make a `testapp.get` request to batch_download endpoint, using the URL that would be generated by cart, and assert that the file is listed in the returned body. See here for framework:
https://github.com/ENCODE-DCC/encoded/blob/5c2b446a05a969ac89460107810d9fa010ad41bd/src/encoded/tests/test_batch_download.py#L109

There's already a bed3+ file (ENCFF880XNW) in data inserts: https://github.com/ENCODE-DCC/encoded/blob/5c2b446a05a969ac89460107810d9fa010ad41bd/src/encoded/tests/data/inserts/file.json#L2572

So you could load workbook in your test, make get request with various file filters included, and assert that ENCFF880XNW is or is not in the body.

`testapp.get('/batch_download/type=Experiment&files.file_type=bed%2Bbed3%252B')`

I would like to see this with several different filters applied. For example `&files.accession=ENCFF880XNW` or `&files.accession!=ENCFF880XNW` (in which case the file shouldn't be in the TSV). 

By the way will negation (!=) work here!?

Reply:
I feel like overall batch_download fixes shouldn’t be in a ticket about cart facets. This fix is in there so that the number of files downloaded isn’t completely different from the number of files promised by the front-end facet code. Unrelated issues should go into one or more other tickets.

Reply:
> Also I don't think the files_prop_param_list function call in the parametrize decorator is a good pattern to use.

I think this should also be addressed in another ticket so that `test_restricted_files_present` can be fixed as well.

Reply:
I agree with @forresttanaka here; it was karthik's weird pattern and outside the scope of this ticket.

I will add your comments to the batch_download test ticket.